### PR TITLE
FIX #1914 Reject command lines with trailing flags

### LINKED ISSUE #1914: `create` silently ignores trailing options
Hi,

I spent most of my day debugging a problem caused by specifying trailing options on `docker-machine create --driver virtualbox default <more options>`, it'd be nice to have some sort of warning message when there are extra arguments specified to alert the user that it will not perform the desired behaviour.

### PR DESCRIPTION
**Pull request by dgageot**

### CODE REVIEW

**Comment by dmp42** [comment]
@dgageot some integration tests are apparently relying on trailing flags - you will have to update them (see `test/integration`)

**Comment by dmp42** [comment]
Also, github will magically link to an issue if fix #1914 is in the body of a comment - unfortunately, not in the title of the PR.

**Comment by dgageot** [comment]
@nathanleclaire @dmp42 wdyt?

File: test/integration/cli/create-rm.bats [hunk 1]
```diff
@@ -53,9 +53,9 @@ load ${BASE_TEST_DIR}/helpers.bash
 }
 
 @test "none: extraneous argument is ignored in name 'machine create -d none --url none a foo'" {
```
Comment (dmp42) [inline test/integration/cli/create-rm.bats]: Shall this test be renamed?

**Comment by dmp42** [comment]
Looks good.
@nathanleclaire can you take a quick look?

**Comment by dgageot** [comment]
I will change the name!
Le sam. 10 oct. 2015 à 21:07, Olivier Gambier <notifications@github.com> a
écrit :

**Comment by dgageot** [comment]
@dmp42 test name fixed. If you or @nathanleclaire could take a look, that would be awsome

**Comment by dmp42** [comment]
LGTM

ping @nathanleclaire can you take a look?

File: commands/create_test.go [hunk 2]
```diff
@@ -20,3 +21,19 @@ func TestValidateSwarmDiscoveryAcceptsValidFormat(t *testing.T) {
 	err := validateSwarmDiscovery("token://deadbeefcafe")
 	assert.NoError(t, err)
 }
+
+func TestRejectCreateTrailingOptions(t *testing.T) {
```
Comment (nathanleclaire) [inline commands/create_test.go]: This seems to be more of an integration test than a unit test (it tests the whole flow), so my preference would be to leave it out.

**Comment by nathanleclaire** [comment]
Why change the `cmdCreate` method so much?  I have made some pretty serious changes in the whole `create` flow for the plugins so I'd prefer the changes to be minimal for now.

File: commands/create_test.go (hunk 2 above)
Comment (dgageot) [inline commands/create_test.go]: I strongly disagree @nathanleclaire. We should test most error cases without integration tests. This test might me a little complicated for now but it's not an integration test. It's just that the design has to be improved to make testing easier.

Until we do those improvements, that's one more test on a file that lacks tests

**Comment by dgageot** [comment]
I changed `create` to make it testable. Having a dozen of `log.Fatal` is plain non-sense test-wise. It makes it impossible to unit-test.

File: test/integration/cli/create-rm.bats [hunk 3]
```diff
@@ -52,10 +52,10 @@ load ${BASE_TEST_DIR}/helpers.bash
   [[ ${lines[0]} == "Error creating machine: Machine A already exists" ]]
 }
 
-@test "none: extraneous argument is ignored in name 'machine create -d none --url none a foo'" {
-  run machine create -d none a foo
+@test "none: fail with trailing options 'machine create -d none --url none a foo'" {
+  run machine create -d none --url none a foo
   [ "$status" -eq 1 ]
-  [[ ${lines[0]} == "Host already exists: \"a\"" ]]
+  [[ ${lines[0]} == "Invalid command line. All options should be either before or after the machine name. Found those trailing flags [foo]" ]]
```
Comment (nathanleclaire) [inline test/integration/cli/create-rm.bats]: This error message doesn't make much sense to me as `foo` is not a flag, it's an extra argument.

Comment (dgageot) [inline test/integration/cli/create-rm.bats]: Good point. I'll change the error message

**Comment by dgageot** [comment]
@nathanleclaire I removed the modifications on `cmdCreate` since it's a pain for your PR. I still believe using `log.Fatal()` like this makes the code impossible to test with unit-tests so I'll create a PR later to address this issue.

I also fixed the wording.

Do you see any issue now?

**Comment by nathanleclaire** [comment]
>  I removed the modifications on cmdCreate since it's a pain for your PR. I still believe using log.Fatal() like this makes the code impossible to test with unit-tests so I'll create a PR later to address this issue.

Yeah, I agree with this, and actually with the new plugins work anything we can do to reduce the possibility of a non-expected exit from the program is a good thing.

**Comment by dgageot** [comment]
@nathanleclaire cool. So I guess it's ready to merge right? cc @nathanleclaire @dmp42

**Comment by dmp42** [comment]
Thanks @nathanleclaire ! 
And thanks @dgageot 

LGTM

**Pull request merged**